# local development docker compose
services:
  reverse-proxy:
    image: traefik:v3.1.2
    command: >
      --api.insecure=true
      --providers.docker

    # --tracing.otlp=true
    # --tracing.otlp.http=true
    # --tracing.otlp.http.endpoint=http://jaeger:4318/v1/traces
    expose:
      - 80
      - 443
    ports:
      - "80:80"
      - "${PROXY_UI_PORT:-8084}:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
  #   depends_on:
  #     - jaeger
  #   environment:
  #     - OTEL_PROPAGATORS=b3,jaeger
  #
  # jaeger:
  #   image: jaegertracing/all-in-one:1.27
  #   ports:
  #     - "5775:5775/udp"
  #     - "6831:6831/udp"
  #     - "6832:6832/udp"
  #     - "14250:14250"
  #     - "16686:16686"
  #     - "14268:14268"
  #     - "4318:4318" # OpenTelemetry Default HTTP Port
  #     - "9411:9411"
  thaumaturgy-python:
    build:
      context: ./thaumaturgy-python/
      dockerfile: ./Dockerfile
    env_file:
      - config/global.env
    environment:
      - MILVUS_DEV_DIR=/tmp/milvus/
    volumes:
      - ./thaumaturgy-python/:/app
      - ${VOLUMES_DIRECTORY:-./volumes}/instance:/app/instance
      - ${VOLUMES_DIRECTORY:-./volumes}/files:/files
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.thaumaturgy-python.rule=PathPrefix(`/alchemy/v1/`) "
    command:
      - "uvicorn"
      - "main:app"
      - "--host"
      - "0.0.0.0"
      - "--port"
      - "5505"
      - "--workers"
      - "4"
      - "--reload"
    expose:
      - 5505

  valkey:
    hostname: valkey
    image: valkey/valkey:7.2.5
    volumes:
      - ./volumes/valkey.conf:/etc/valkey/valkey.conf
      - ./volumes/valkey-data:/data
    command: valkey-server /etc/valkey/valkey.conf
    healthcheck:
      test: ["CMD-SHELL", "redis-cli ping | grep PONG"]
      interval: 1s
      timeout: 3s
      retries: 5
    expose:
      - 6379
    ports:
      - 6379:6379
